pkg/actuators/client/client.go

```diff
@@ -400,6 +409,28 @@ func (c *ibmCloudClient) GetCustomImageByName(imageName string, resourceGroupID
 	return "", fmt.Errorf("could not retrieve image id of name: %v", imageName)
 }
 
+// VerifyInstanceProfile verifies the provided instance profile exists
+func (c *ibmCloudClient) VerifyInstanceProfile(profileName string) (string, error) {
+	// Get list of instance profiles
+	instanceProfilesList, _, err := c.vpcService.ListInstanceProfiles(c.vpcService.NewListInstanceProfilesOptions())
+	if err != nil {
+		return "", err
+	}
+
+	if instanceProfilesList != nil {
+		for _, instanceProfile := range instanceProfilesList.Profiles {
+			if *instanceProfile.Name == profileName {
+				return profileName, nil
+			}
+		}
+		return "", &machine.MachineError{
```

Comment:
You can use https://github.com/openshift/machine-api-operator/blob/128c5c90918c009172c6d24d5715888e0e1d59e4/pkg/controller/machine/errors.go#L43 as a helper here

Reply:
Okay, I'll update

Reply:
Updated